kong/db/schema/init.lua

```diff
@@ -1702,6 +1703,28 @@ local function collect_field_reference(refs, key, reference)
   return refs
 end
 
+local function validate_deprecation_exclusiveness(data, shorthand_value, shorthand_name, shorthand_definition)
+  if shorthand_definition.deprecation and shorthand_definition.deprecation.replaced_with then
+    for _, replaced_with_path in ipairs(shorthand_definition.deprecation.replaced_with) do
+      local new_field_value = table_path(data, replaced_with_path)
+
+      if new_field_value ~= nil and
+          new_field_value ~= null and
```

Comment:
Is there reason for the `null` being ignored?

Reply:
You're right. There might be an intentional nil sent by a user so it shouldn't be ignored.

Reply:
Done ✅ 
I've added some tests to verify what happens when someone send `null` over json which translates to `ngx.null`